extensions/interactions/GraphInput/directives/graph-viz.component.ts

```diff
@@ -0,0 +1,626 @@
+  @Input() canDeleteEdge: boolean;
+  @Input() canEditEdgeWeight: boolean;
+  @Input() interactionIsActive: boolean;
+  @Input() canEditOptions: boolean;
+  isMobile: boolean = false;
+  helpText: string = '';
+  _MODES = {
+    MOVE: 0,
+    ADD_EDGE: 1,
+    ADD_VERTEX: 2,
+    DELETE: 3
+  };
+  // Styling functions.
+  DELETE_COLOR = 'red';
+  HOVER_COLOR = 'aqua';
+  SELECT_COLOR = 'orange';
+  DEFAULT_COLOR = 'black';
+  state = {
+    currentMode: this._MODES.MOVE,
+    // Vertex, edge, mode button, label currently being hovered over.
+    hoveredVertex: null,
+    hoveredEdge: null,
+    hoveredModeButton: null,
+    // If in ADD_EDGE mode, source vertex of the new edge, if it
+    // exists.
+    addEdgeVertex: null,
+    // Currently dragged vertex.
+    currentlyDraggedVertex: null,
+    // Selected vertex for editing label.
+    selectedVertex: null,
+    // Selected edge for editing weight.
+    selectedEdge: null,
+    // Mouse position in SVG coordinates.
+    mouseX: 0,
+    mouseY: 0,
+    // Original position of dragged vertex.
+    vertexDragStartX: 0,
+    vertexDragStartY: 0,
+    // Original position of mouse when dragging started.
+    mouseDragStartX: 0,
+    mouseDragStartY: 0
+  };
+  selectedEdgeWeightValue: number | string;
+  buttons: {text: string, description: string, mode: number}[] = [];
+  private vizContainer: SVGSVGElement[];
+  componentSubscriptions: Subscription = new Subscription();
+  shouldShowWrongWeightWarning: boolean;
+  VERTEX_RADIUS: number;
+  EDGE_WIDTH: number;
+  vizWidth: SVGAnimatedLength;
+  graphOptions: { text: string; option: string; }[];
+  svgViewBox: string;
+  constructor(
+    private deviceInfoService: DeviceInfoService,
+    private element: ElementRef,
+    private focusManagerService: FocusManagerService,
+    private graphDetailService: GraphDetailService,
+    private playerPositionService: PlayerPositionService,
+    private utilsService: UtilsService
+  ) {}
+
+  ngOnInit(): void {
+    this.componentSubscriptions.add(
+      this.playerPositionService.onNewCardAvailable.subscribe(
+        () => this.state.currentMode = null
+      )
+    );
+    this.VERTEX_RADIUS = this.graphDetailService.VERTEX_RADIUS;
+    this.EDGE_WIDTH = this.graphDetailService.EDGE_WIDTH;
+    this.selectedEdgeWeightValue = 0;
+    this.shouldShowWrongWeightWarning = false;
+
+    this.isMobile = false;
+    if (this.deviceInfoService.isMobileDevice()) {
+      this.isMobile = true;
+    }
+  }
+
+  ngAfterViewInit(): void {
+    this.vizContainer = this.element.nativeElement.querySelectorAll(
+      '.oppia-graph-viz-svg');
+    this.vizWidth = this.vizContainer[0].width;
+
+    this.graphOptions = [{
+      text: 'Labeled',
+      option: 'isLabeled'
+    },
+    {
+      text: 'Directed',
+      option: 'isDirected'
+    },
+    {
+      text: 'Weighted',
+      option: 'isWeighted'
+    }];
+    this.helpText = null;
+    $(document).on('mouseup', () => this.onMouseupDocument());
```

Comment:
Is this a jQuery code? Should we get rid of this?

Reply:
Done!